lib/common/output_log.c

```diff
@@ -0,0 +1,357 @@
+    }
+    g_queue_free(priv->messages);
+    free(priv);
+}
+
+static bool
+log_init(pcmk__output_t *out) {
+
+    /* If log_init was previously called on this output struct, just return. */
+    if (out->priv != NULL) {
+        return true;
+    }
+
+    out->priv = calloc(1, sizeof(private_data_t));
+    if (out->priv == NULL) {
+         return false;
+     }
+
+    ((private_data_t *)out->priv)->attr.log_level = LOG_INFO;
+    ((private_data_t *)out->priv)->messages = g_queue_new();
+    return true;
+}
+
+static void
+log_finish(pcmk__output_t *out, crm_exit_t exit_status, bool print, void **copy_dest) {
+    /* This function intentionally left blank */
+}
+
+static void
+log_reset(pcmk__output_t *out) {
+    CRM_ASSERT(out && out->priv);
+
+    log_free_priv(out);
+    log_init(out);
+}
+
+G_GNUC_PRINTF(2, 3)
+static void
+pcmk__output_crm_log(pcmk__output_t *out, const char *format, ...) {
+    int len = 0;
+    va_list ap;
+    private_log_message_t *message = calloc(1, sizeof(private_log_message_t));
+    const private_data_t *priv = NULL;
+    CRM_ASSERT(message != NULL);
+    CRM_ASSERT(out && out->priv);
+    priv = (private_data_t *)out->priv;
+
+    message->attr.log_level = priv->attr.log_level;
+    strcpy(message->attr.filename, priv->attr.filename);
+    strcpy(message->attr.function, priv->attr.function);
+    message->attr.lineno = priv->attr.lineno;
+
+    va_start(ap, format);
+    len = vasprintf(&message->text, format, ap);
+    CRM_ASSERT(len >= 0);
+    va_end(ap);
+
+    /* save the message for future use */
+    g_queue_push_tail(priv->messages, message);
+    /* and print it now */
+    log_output_message(message);
+}
+
+static void
+log_version(pcmk__output_t *out, bool extended) {
+    /* Make sure the out->attr is already set */
+    if (extended) {
+        pcmk__output_crm_log(out, "Pacemaker %s (Build: %s): %s\n",
+                   PACEMAKER_VERSION, BUILD_VERSION, CRM_FEATURES);
+    } else {
+        pcmk__output_crm_log(out, "Pacemaker %s\n", PACEMAKER_VERSION);
+        pcmk__output_crm_log(out, "Written by Andrew Beekhof\n");
+    }
+}
+
+G_GNUC_PRINTF(2, 3)
+static void
+log_err(pcmk__output_t *out, const char *format, ...) {
+    va_list ap;
+    char* buffer = NULL;
+    int len = 0;
+    uint8_t log_level;
+    private_data_t *priv = NULL;
+
+    CRM_ASSERT(out && out->priv);
+    va_start(ap, format);
+
+    /* Informational output does not get indented, to separate it from other
+     * potentially indented list output.
+     */
+    len = vasprintf(&buffer, format, ap);
+    CRM_ASSERT(len >= 0);
+    va_end(ap);
+
+    priv = (private_data_t *)out->priv;
+    /* backup the log_level*/
+    log_level = priv->attr.log_level;
+    priv->attr.log_level = LOG_ERR;
+    /* print out the log message with the LOG_ERR severity */
+    pcmk__output_crm_log(out, "%s", buffer);
+    /* restore the log_level backup */
+    priv->attr.log_level = log_level;
+    free(buffer);
```

Comment:
This function can keep using LOG_ERR for obvious reasons, but it should be sufficient to just call crm_err() directly

Reply:
Done. The crm_err is used instead.